### ISSUE TITLE
[macOS & Linux] External: Build failure caused by updating to fmt 10

### ISSUE THREAD

**Issue by shinra-electric** [yuzu-emu/yuzu#10410]
### Issue

This is likely an external issue, and not a regression with Yuzu. 

I was building Yuzu fine for the last few weeks, but suddenly I couldn't open any previously-compiled builds, and was unable to compile a fresh one. 

When opening an existing build, I got the error that fmt.9.dylib couldn't be found. So I checked my installs, and I had version 10, which was released a couple of weeks ago. I'm guessing that the builds started failing after I installed the update, but I can't be 100% sure. (I usually just use the `brew upgrade` command once in a while and blindly update whatever is available)

I tried building Yuzu locally, and reverted several PRs to see if they were the cause of the issue, but I went back a couple of weeks and it still failed, so I am fairly sure it wasn't a change in Yuzu that caused this.

[Terminal Saved Output.txt](https://github.com/yuzu-emu/yuzu/files/11531914/Terminal.Saved.Output.txt)

Unfortunately reverting fmt to version 9.1 is not easy, and I haven't been able to do it so far. I also can't install multiple versions like I would with Qt or LLVM, since fmt doesn't work that way. 

It would be helpful if someone who still has fmt version 9.1 installed can confirm that updating is it the cause of this issue.

**Comment by liamwhite** [comment]
Do you have a log of the compilation failure?

**Comment by shinra-electric** [comment]
Added the log above.

**Comment by hrzhu** [comment]
Building yuzu-mainline against fmt 10  fails on Ubuntu 22.04 too.

```[1154/1862] Building CXX object src/co...e.dir/hle/service/audio/audren_u.cpp.o
FAILED: src/core/CMakeFiles/core.dir/hle/service/audio/audren_u.cpp.o 
/usr/bin/g++-12 -DARCHITECTURE_x86_64=1 -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DENABLE_WEB_SERVICE -DFMT_SHARED -DNDEBUG -DOPUS_VERSION="\"\\\"1.3.1\\\"\"" -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/home/h/src/yuzu-mainline/src/. -I/home/h/src/yuzu-mainline/externals/./microprofile -I/home/h/src/yuzu-mainline/externals/glad/include -I/home/h/src/yuzu-mainline/externals/sirit/src/../include -I/home/h/src/yuzu-mainline/externals/sirit/externals/SPIRV-Headers/include -I/home/h/src/yuzu-mainline/externals/xbyak -I/home/h/src/yuzu-mainline/externals/./mbedtls/include -I/home/h/src/yuzu-mainline/externals/opus/opus/include -I/home/h/src/yuzu-mainline/externals/dynarmic/src/dynarmic/.. -isystem /home/h/bin/deps/include -march=native -O3 -pipe -O3 -DNDEBUG -flto -fno-fat-lto-objects -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -mcx16 -fwrapv -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -Werror=conversion -Wno-sign-conversion -std=gnu++20 -Winvalid-pch -include /home/h/src/yuzu-mainline/build-fmt10/src/core/CMakeFiles/core.dir/cmake_pch.hxx -MD -MT src/core/CMakeFiles/core.dir/hle/service/audio/audren_u.cpp.o -MF src/core/CMakeFiles/core.dir/hle/service/audio/audren_u.cpp.o.d -o src/core/CMakeFiles/core.dir/hle/service/audio/audren_u.cpp.o -c /home/h/src/yuzu-mainline/src/core/hle/service/audio/audren_u.cpp
In file included from /home/h/bin/deps/include/fmt/format.h:49,
                 from /home/h/src/yuzu-mainline/src/./common/common_precompiled_headers.h:11,
                 from /home/h/src/yuzu-mainline/src/core/precompiled_headers.h:9,
                 from /home/h/src/yuzu-mainline/build-fmt10/src/core/CMakeFiles/core.dir/cmake_pch.hxx:5,
                 from <command-line>:
/home/h/bin/deps/include/fmt/core.h: In instantiation of ‘constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_value(T&&) [with Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; T = const BitField<9, 13, unsigned int>&]’:
/home/h/bin/deps/include/fmt/core.h:1711:29:   required from ‘constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_arg(T&&) [with bool IS_PACKED = true; Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; type <anonymous> = fmt::v10::detail::type::custom_type; T = const BitField<9, 13, unsigned int>&; typename std::enable_if<IS_PACKED, int>::type <anonymous> = 0]’
/home/h/bin/deps/include/fmt/core.h:1827:77:   required from ‘constexpr fmt::v10::format_arg_store<Context, Args>::format_arg_store(T&& ...) [with T = {const BitField<9, 13, unsigned int, KeepTag>&}; Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; Args = {BitField<9, 13, unsigned int, KeepTag>}]’
/home/h/bin/deps/include/fmt/core.h:1844:31:   required from ‘constexpr fmt::v10::format_arg_store<Context, typename std::remove_cv<typename std::remove_reference<T>::type>::type ...> fmt::v10::make_format_args(T&& ...) [with Context = basic_format_context<appender, char>; T = {const BitField<9, 13, unsigned int, KeepTag>&}]’
/home/h/src/yuzu-mainline/src/./common/logging/log.h:34:44:   required from ‘void Common::Log::FmtLogMessage(Class, Level, const char*, unsigned int, const char*, const char*, const Args& ...) [with Args = {BitField<9, 13, unsigned int, KeepTag>}]’
/home/h/src/yuzu-mainline/src/core/hle/service/audio/audren_u.cpp:143:13:   required from here
/home/h/bin/deps/include/fmt/core.h:1691:7: error: static assertion failed: Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
 1691 |       formattable,
      |       ^~~~~~~~~~~
/home/h/bin/deps/include/fmt/core.h:1691:7: note: ‘formattable’ evaluates to false
[1169/1862] Building CXX object src/co...le/service/filesystem/filesystem.cpp.o
ninja: build stopped: subcommand failed.
```

**Comment by yingnierxiao** [comment]
wget  https://raw.githubusercontent.com/Homebrew/homebrew-core/199127ec9184b3be2d6d3a17a3bfc2fd13c6c39a/Formula/fmt.rb
brew uninstall fmt 
brew install fmt.rb  
uninstall fmt 10,and install fmt9

**Comment by shinra-electric** [comment]
Thanks to Yingnierxiao I have reverted to fmt 9.1 and am able to build and run Yuzu normally. 

This confirms that fmt 10 is the cause of the issue.

**Comment by shinra-electric** [comment]
This will probably be fixed by #10484
